src/firefly/js/Firefly.js

```diff
@@ -365,17 +368,17 @@ function bootstrap(props, clientAppSpecificOptions, webApiCommands) {
     });
 }
 
-function renderRoot(viewer, props, webApiCommands) {
+function renderRoot(root, viewer, props, webApiCommands) {
     const e= document.getElementById(props.div);
     if (!e) {
         showInfoPopup('HTML page is not setup correctly, Firefly cannot start.');
         logger.error(`DOM Element "${props.div}" is not found in the document, Firefly cannot start.`);
         return;
     }
 
-
+    const rootToUse = root ?? createRoot(e);
     const webApi= isUsingWebApi(webApiCommands);
-    const doAppRender= () => ReactDOM.render(React.createElement(viewer, {...props, normalInit: !webApi }), e);
+    const doAppRender= () => rootToUse.render( React.createElement(viewer, {...props, normalInit: !webApi }) );
```

Comment:
Have you tried add an ErrorBoundary around the 'viewer' to see if it'll catch all uncaught errors?
It may not work as expected, but it's worth a try.  There's very little code needed to make this happen.

```js
const app = React.createElement(viewer,   ...
rootToUse.render(
  <ErrorBoundary>
    {app}
  </ErrorBoundary>,
  
  ...  
```

Reply:
I was looking for `ErrorBoundary` but could not find it in the docs.

Reply:
I still can't fine it. Can you point me to the docs?

Reply:
As far as I can tell this is only documented in the legacy API my creating such a component using `componentDidCatch()`, but this is in what they now call their Legacy React API.

Reply:
I have no idea whether or not an uncaught exception would propagate to the root rendering.  I was only curious to see if you've tried.  Below is a 'class' based implementation.  There's a functional version as well.  If it does, then we can at least intercept it and provide messages with some sort of reset link.

https://legacy.reactjs.org/docs/error-boundaries.html
```js
class ErrorBoundary extends React.Component {
  constructor(props) {
    super(props);
    this.state = { hasError: false };
  }

  static getDerivedStateFromError(error) {
    // Update state so the next render will show the fallback UI.
    return { hasError: true };
  }

  componentDidCatch(error, errorInfo) {
    // You can also log the error to an error reporting service
    logErrorToMyService(error, errorInfo);
  }

  render() {
    if (this.state.hasError) {
      // You can render any custom fallback UI
      return <h1>Something went wrong.</h1>;
    }

    return this.props.children; 
  }
}```

Reply:
yes, this is what I have found as well. This was added in react 16. They are now acting like it is 'legacy'. Either way it is out of scope for this PR. We could have put this in for the last 2 years. I think it is a different task from the upgrade.